patchwork-extensions-block/src/main/java/net/patchworkmc/mixin/extensions/block/blockentity/MixinSpongeBlock.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.extensions.block.blockentity;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.SpongeBlock;
+
+import net.patchworkmc.impl.extensions.block.BlockContext;
+import net.patchworkmc.impl.extensions.block.Signatures;
+
+@Mixin(SpongeBlock.class)
+public abstract class MixinSpongeBlock {
+	@Redirect(method = "absorbWater", at = @At(value = "INVOKE", target = Signatures.BlockState_getBlock, ordinal = 3))
```

Comment:
3 for an ordinal is a bit high. Can you use an `@Slice(from=@At(...))` here to bring that down?

Reply:
They are located very closely with each other in a series of if statements are they are the only occurrence in this method. If we look at the code of the target method, it is very straightforward to tell the ordinal. Adding `@Slice(from..to..)` might introduce unnecessary complexity.